src/audio/SDL_audiocvt.c

```diff
@@ -233,6 +237,23 @@ void ConvertAudio(int num_frames, const void *src, SDL_AudioFormat src_format, i
         src = buf;
     }
 
+    // Gain adjustment
+    if (gain != 1.0f) {
+        float *buf = (float *) (dstconvert ? scratch : dst);
```

Comment:
```suggestion
        float *buf = (float *)(dstconvert ? scratch : dst);
```

Reply:
Fixed in c1a3f6d0eccb6dc5aa9bebf20f982a02cee946f1.